### ISSUE TITLE
Pytorch LR Scheduler Integration

### ISSUE THREAD

**Issue by juan43ramirez** [cooper-org/cooper#17]
**Context:** Many current training pipelines involve tweaking the step size/learning rate throughout training to achieve good performance. Two common scenarios for this are:
- Decreasing the learning rate can help to get closer to the optimum, rather than keep bouncing around it.
- It can be useful when considering problems in which the theoretical guarantees involved an assumption on diminishing step sizes.

**Proposal:** Add capabilities to use [Pytorch's LR schedulers](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate) along with Cooper.

**Challenges:** 
- It should be relatively straightforward to handle this for the  `primal_optimizer` as it is "fully instantiated" by the user before creating the `ConstrainedOptimizer`. 
- It gets a bit tricky with the `dual_optimizer` as we don't have access to the full optimizer before the Lagrange multipliers have been initialized.
- One important thing here is to keep the definition of the lr scheduler as consistent as possible with that of the dual optimizer: since we ask for a "partially instantiated optimizer" maybe we should ask for a "partially instantiated scheduler"?
- Shall we just leave it under the user's control (just as in Pytorch) to perform the calls to `lr_scheduler.step()`?

_Originally posted by @gallego-posada in https://github.com/cooper-org/cooper/discussions/16_

**Comment by juan43ramirez** [comment]
Accidentally created a duplicate to #16 . Closing
